auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -144,4 +170,91 @@ public BidSummaryResponse getBidSummary(final UUID productId) {
 
         return new BidSummaryResponse(productEntity.getHighestBid(), productEntity.getBidsCount());
     }
+
+    @Override
+    public Page<ProductBidDetailsResponse> getProductByUserAndStatus(final UUID userId,
+                                                                     final ProductStatus productStatus,
+                                                                     final int page,
+                                                                     final int size) {
+        Pageable pageable = PageRequest.of(page, size);
+
+        return this.productRepository
+                .findProductEntityByUserEntity_UserIdAndAndStatus(userId, productStatus, pageable)
+                .map(productEntity -> {
+                    ProductBidDetailsResponse response = new ProductBidDetailsResponse();
```

Comment:
Why don't we have method to map this, like with toDomainModel, to avoid this much code